internal/services/gitserver/repo-cleaner.go

```diff
@@ -0,0 +1,167 @@
+package gitserver
+
+import (
+	"context"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"agola.io/agola/internal/util"
+)
+
+func (s *Gitserver) repoCleanerLoop(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			log.Info("repoCleaner exiting")
+
+			return
+		case <-time.After(s.c.RepositoryCleanupInterval):
+			if err := s.scanRepos(ctx); err != nil {
+				log.Errorf("err: %+v", err)
+			}
+		}
+	}
+}
+
+func (s *Gitserver) scanRepos(ctx context.Context) error {
+	log.Info("repoCleaner scanRepos start")
+
+	usersDir, err := ioutil.ReadDir(s.c.DataDir)
+	if err != nil {
+		return err
+	}
+
+	for _, u := range usersDir {
+		if !u.IsDir() {
+			continue
+		}
+
+		reposDir, _ := ioutil.ReadDir(filepath.Join(s.c.DataDir, u.Name()))
+		for _, r := range reposDir {
+			if r.IsDir() {
+				s.scanRepo(ctx, filepath.Join(s.c.DataDir, u.Name(), r.Name()))
+			}
+		}
+	}
+
+	log.Info("repoCleaner scanRepos end")
+
+	return nil
+}
+
+func (s *Gitserver) scanRepo(ctx context.Context, repoDir string) {
+	git := &util.Git{GitDir: repoDir}
+
+	branches, _ := s.getBranches(git, ctx)
+	deletedCount := 0
+	for _, b := range branches {
+		date, err := s.getLastCommitTime(ctx, git, "refs/heads/"+b)
+		if err != nil {
+			log.Errorf("repoCleaner error:", err)
+			continue
+		}
+
+		if time.Since(date) >= s.c.RepositoryRefsExpireInterval {
+			err := s.deleteBranch(ctx, git, b)
+			if err != nil {
+				log.Errorf("repoCleaner error: %v", err)
+			} else {
+				deletedCount++
+			}
+		}
+	}
+
+	tags, _ := s.getTags(git, ctx)
+	for _, tag := range tags {
+		date, err := s.getLastCommitTime(ctx, git, "refs/tags/"+tag)
+		if err != nil {
+			log.Error("repoCleaner error:", err)
+			continue
+		}
+
+		if time.Since(date) >= s.c.RepositoryRefsExpireInterval {
+			err := s.deleteTag(ctx, git, tag)
+			if err != nil {
+				log.Error("repoCleaner error:", err)
```

Comment:
Remove logging and just use `return err` or `return fmt.Errorf("failed to delete git tag: %w", err)`